lib/iris/analysis/_grid_angles.py

```diff
@@ -0,0 +1,390 @@
+# (C) British Crown Copyright 2010 - 2018, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Code to implement vector rotation by angles, and inferring gridcell angles
+from coordinate points and bounds.
+
+"""
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+import numpy as np
+
+import iris
+
+
+def _3d_xyz_from_latlon(lon, lat):
+    """
+    Return locations of (lon, lat) in 3D space.
+
+    Args:
+
+    * lon, lat: (arrays in degrees)
+
+    Returns:
+
+        xyz : (array, dtype=float64)
+            cartesian coordinates on a unit sphere.  Dimension 0 maps x,y,z.
+
+    """
+    lon1 = np.deg2rad(lon).astype(np.float64)
+    lat1 = np.deg2rad(lat).astype(np.float64)
+
+    x = np.cos(lat1) * np.cos(lon1)
+    y = np.cos(lat1) * np.sin(lon1)
+    z = np.sin(lat1)
+
+    result = np.concatenate([array[np.newaxis] for array in (x, y, z)])
+
+    return result
+
+
+def _latlon_from_xyz(xyz):
+    """
+    Return arrays of lons+lats angles from xyz locations.
+
+    Args:
+
+    * xyz: (array)
+        positions array, of dims (3, <others>), where index 0 maps x/y/z.
```

Comment:
This is a bit confusing.  I know that the docs in general are not finished though, so I won't go on.